cmake/OpenCVDetectCUDA.cmake

```diff
@@ -75,21 +84,39 @@ if(CUDA_FOUND)
     if(WITH_NVCUVID)
       ocv_cuda_SEARCH_NVCUVID_HEADER("nvcuvid.h" HAVE_NVCUVID_HEADER)
       ocv_cuda_SEARCH_NVCUVID_HEADER("dynlink_nvcuvid.h" HAVE_DYNLINK_NVCUVID_HEADER)
+      if(DEFINED CUDA_nvcuvid_LIBRARY_INTERNAL)
+        set(CUDA_nvcuvid_LIBRARY ${CUDA_nvcuvid_LIBRARY_INTERNAL} CACHE FILEPATH "\"nvcuvid\" library" FORCE)
```

Comment:
> calls to FindCUDA can reset

Should we check for the empty value then?

Reply:
@alalek  I agree but I've just noticed this issue was fixed in CMake 6 months ago https://gitlab.kitware.com/cmake/cmake/-/commit/a7758394afc2717bb31a47826e62c1ba86a63c41.  I suggest for compatibility with older versions of CMake we could apply the same fix they used which I mentioned above and simply ensure that `CUDA_TOOLKIT_TARGET_DIR` is set before calling FindCUDA what do you think?